### PR TITLE
[Merged by Bors] - Angular/hal fix

### PR DESCRIPTION
**Pull request by mikolaichuk**
## Issue
https://app.clickup.com/t/2rbudp3
https://app.clickup.com/t/2rq1arp

## Description
This branch accumulates changes which were done as per Hallborn validation report. 
We decided to merge these changes in the main before Hallborn's review to avoid conflicts. 
This changes includes Dzmitry's suggestions from this PR: https://github.com/ComposableFi/composable/pull/1221,
To fix nit regarding weight of xcm_sell() dispatchable function:  https://github.com/ComposableFi/composable/pull/1221#discussion_r909367212
corresponded benchmarks were implemented to change static weights to dynamic ones (for xcm_sell() and add_configuration() dispatchable functions).
## Checklist
- [ ] I have updated the cargo docs to reflect changes made by this PR _(if applicable)_
- [ ] I have updated the `book/` to reflect changes made by this PR _(if applicable)_

### CODE REVIEW

File: frame/dutch-auction/src/benchmarking.rs [hunk 1]
```diff
@@ -1,16 +1,23 @@
...
 use frame_support::traits::{fungibles::Mutate, Currency, Get, Hooks};
 use frame_system::{pallet_prelude::BlockNumberFor, RawOrigin};
+use sp_core::{crypto::UncheckedFrom, H256};
 use sp_runtime::{
 	traits::{AccountIdConversion, Saturating},
 	FixedPointNumber,
 };
 use sp_std::prelude::*;
 
+type TraitBalance<T> = <T as DeFiComposableConfig>::Balance;
```
Comment (KaiserKarel) [inline frame/dutch-auction/src/benchmarking.rs:19]: Prefer this to be `BalanceOf<T>`

**Review by benluelo — APPROVED** [review]

File: frame/liquidations/src/mock/runtime.rs [hunk 2]
```diff
@@ -231,19 +233,21 @@ impl pallet_dutch_auction::Config for Runtime {
 
 parameter_types! {
 	pub const LiquidationPalletId : PalletId = PalletId(*b"liqudatn");
+	pub const MaxLiquidationStrategiesAmount: u32 = 3;
```
[the remark this answers is not in the document]
Comment (mikolaichuk) [reply frame/liquidations/src/mock/runtime.rs]: Fixed

File: frame/liquidations/src/lib.rs [hunk 3]
```diff
@@ -259,7 +276,7 @@ pub mod pallet {
 							T::DutchAuction::ask(from_to, order.clone(), configuration),
 						_ =>
 							return Err(DispatchError::Other(
-								"as for now, only auction liquidators implemented",
+								"As for now, only auction liquidation strategy is implemented",
```
[the remark this answers is not in the document]
Comment (mikolaichuk) [reply frame/liquidations/src/lib.rs]: Fixed

File: frame/lending/src/mocks/general.rs [hunk 4]
```diff
@@ -337,16 +337,21 @@ impl pallet_dutch_auction::Config for Runtime {
 	type XcmSender = XcmFake;
 }
 
+parameter_types! {
+	pub const MaxLiquidationStrategiesAmount: u32 = 3;
```
[the remark this answers is not in the document]
Comment (mikolaichuk) [reply frame/lending/src/mocks/general.rs]: Fixed

**Review by benluelo — APPROVED** [review]

**Review by dzmitry-lahoda — APPROVED** [review]
assuming logic of moved stuff was not changed

**Comment by mikolaichuk** [comment]
bors merge

**Review by benluelo — APPROVED** [review]

**Review by kevin-composable — DISMISSED** [review]

**Review by dzmitry-lahoda — APPROVED** [review]
bors r+

**Review by hussein-aitlahcen — APPROVED** [review]

**Comment by mikolaichuk** [comment]
bors merge

**Comment by mikolaichuk** [comment]
bors merge

**Review by KaiserKarel — APPROVED** [review]

**Comment by mikolaichuk** [comment]
bors merge

**Review by hussein-aitlahcen — APPROVED** [review]

**Review by KaiserKarel — APPROVED** [review]

**Comment by mikolaichuk** [comment]
bors merge

**Review by dzmitry-lahoda — APPROVED** [review]

**Comment by mikolaichuk** [comment]
bors merge

**Pull request closed without merging**